sos/plugins/watchdog.py

```diff
@@ -0,0 +1,90 @@
+# Copyright (C) 2018 Red Hat, Inc., Reid Wahl <nwahl@redhat.com>
+
+# This program is free software; you can redistribute it and/or modify
+# it under the terms of the GNU General Public License as published by
+# the Free Software Foundation; either version 2 of the License, or
+# (at your option) any later version.
+
+# This program is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+# GNU General Public License for more details.
+
+# You should have received a copy of the GNU General Public License
+# along with this program; if not, write to the Free Software
+# Foundation, Inc., 675 Mass Ave, Cambridge, MA 02139, USA.
+'''Collect info from watchdog to add to sosreport.
+
+Classes:
+Watchdog -- collects watchdog configs, custom executables, and logs
+'''
+
+import re
+
+from sos.plugins import Plugin, RedHatPlugin
+
+
+class Watchdog(Plugin, RedHatPlugin):
+    '''Collect watchdog configurations, custom executables, and logs.
+
+    Methods:
+    parse_watchdog_conf -- parse variables from config file
+    setup -- assemble watchdog info for sosreport
+
+    Instance variables:
+    limit -- maximum log size
+    watchdog_conf -- watchdog service configuration file
+    conf_vars -- variables from watchdog config file
+    log_dir -- directory containing watchdog stderr and stdout logs
+    '''
+
+    plugin_name = 'watchdog'
+    profiles = ('system',)
+    packages = ('watchdog',)
+
+    def __init__(self, commons):
+        '''Initialize Watchdog class.'''
+        Plugin.__init__(self, commons)
+        self.limit = self.get_option('log_size')
+        self.watchdog_conf = '/etc/watchdog.conf'
+        self.conf_vars = dict()
+        self.log_dir = None
+
+    def parse_watchdog_conf(self):
+        '''Parse variables from watchdog config file.'''
+        comment_pattern = re.compile(r'#.*')
+        eq_pattern = re.compile(r'\s*=\s*')
+
+        with open(self.watchdog_conf, 'r') as conf_file:
+            for line in conf_file:
+                line = comment_pattern.sub('', line)
+                line = line.strip()
+
+                try:
+                    (key, value) = eq_pattern.split(line, 1)
+                    self.conf_vars[key] = value
```

Comment:
Why to store all configuration in `conf_vars`, if just the only `log-dir` is used further? Isn't it easier to grep for files with pattern like `[ ]*log-dir[ ]*=[ ]*$1` (not correct regexp) and store the value?

Reply:
That is a good point. I normally try to take a general, extensible approach to parsing log files, as I've done here. However, I don't foresee any other parameters in `/etc/watchdog.conf` being useful now or in the future in this context.

I can put together an alternate version using only `log-dir` in the morning (it's late here now) if you want, and we can decide which one looks cleaner. Readability is going to trump computational/memory efficiency here IMO, with this being such a small file to parse.

Reply:
As @pmoravec said, there's not much point in fully parsing this file if the results will never be used. The code to simply extract the `log-dir` key and its value is more compact and easier to read.

There's also no need to deploy a regex for comment removal - since the comment is started by `#` you can simply split on that character and discard everything else.

Reply:
You're collecting the file in any case, so the dictionary really is not adding any value at this point.

Reply:
> There's also no need to deploy a regex for comment removal.
Yep, duh... I think I had something more complex initially and dropped it to that.